maple/classification.py

```diff
@@ -0,0 +1,43 @@
+import re
+from typing import TypeVar
+
+from maple.types import Action, ActionType
+
+T = TypeVar("T")
+
+
+def parse(rules: list[tuple[re.Pattern, T]], x: str) -> T | None:
+    """Parse a string according to an ordered sequence of rules.
+
+    The first matching rule will be constructed by calling the
+    matching function with the regex match.
+    """
+
+    for rule, label in rules:
+        if rule.match(x) is not None:
+            return label
+
+    return None
+
+
+def regex_classification(action: Action) -> ActionType:
+    if (t := parse(_action_rules, action.action)) is not None:
```

Comment:
Thar be walruses!  Just a reminder that we need an environment spec because this puts a hard requirement on the python version.

Reply:
1. I thought I saw 3.10 mentioned in some existing requirements file, but maybe I hallucinated that
2. I'm fine removing 3.10+ features if we think we want to run on 3.9 or something